db-service/lib/SQLService.js

```diff
@@ -20,6 +20,39 @@ class SQLService extends DatabaseService {
     this.on(['UPDATE'], this.transformStreamIntoCQN)
     this.on(['INSERT', 'UPSERT', 'UPDATE'], require('./fill-in-keys')) // REVISIT: should be replaced by correct input processing eventually
     this.on(['INSERT', 'UPSERT', 'UPDATE'], require('./deep-queries').onDeep)
+    if (cds.env.features.db_strict) {
+      this.on(['INSERT', 'UPSERT', 'UPDATE'], (req, next) => {
+        const { query } = req
+        const kind = query.kind || Object.keys(query)[0]
+        const operation = query[kind]
+        if(!operation.columns && !operation.entries && !operation.data) return next()
+        const columns =
+          operation.columns ||
+          Object.keys(
+            operation.entries?.reduce((acc, obj) => {
+              return Object.assign(acc, obj)
+            }, {}) || operation.data,
+          )
+        const invalidColumns = columns.filter(c => !query.target?.elements?.[c])
+
+        if (invalidColumns.length > 0) {
+          const createColumnError = (tableName, columnName) =>
+            Object.assign(new Error(`Table ${tableName} has no column named ${columnName}`), { code: 400 })
+
+          const err =
+            invalidColumns.length === 1
+              ? createColumnError(query.target.name, invalidColumns[0])
+              : new Error('MULTIPLE_ERRORS')
+
+          if (invalidColumns.length > 1) {
+            err.details = invalidColumns.map(el => createColumnError(query.target.name, el))
+          }
+
+          throw Object.assign(err, { statusCode: 400 })
```

Comment:
Some how this comment was not included in the last review.

The database service is not allowed to use HTTP codes in the errors. As it is not possible to know what is the source of the erroneous query it cannot be determined that the code should be `400`. Arguably it is least likely that this would happen with queries coming from the protocol adapters as they tent to check their payload before passing it along to the database service. So most likely this error would result in a `500`.

Also the multiple errors occurred error message format used is `OData` specific and would be very confusing when the error happens multiple times in a OData request, because it would say multiple times that multiple errors occurred. 

```suggestion
        if (invalidColumns.length > 0) {
          cds.error(`STRICT MODE: Trying to ${kind} non existent columns (${invalidColumns})`)
```

Reply:
thanks for the explanation. removed it according to your suggestion